x-pack/plugins/cases/public/components/files/file_type.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import type {
+  ExternalReferenceAttachmentType,
+  ExternalReferenceAttachmentViewProps,
+} from '../../client/attachment_framework/types';
+
+import { AttachmentActionType } from '../../client/attachment_framework/types';
+import { FILE_ATTACHMENT_TYPE } from '../../../common/api';
+import { FileDownloadButtonIcon } from './file_download_button_icon';
+import { FileNameLink } from './file_name_link';
+import { FilePreview } from './file_preview';
+import * as i18n from './translations';
+import { isImage, isValidFileExternalReferenceMetadata } from './utils';
+import { useFilePreview } from './use_file_preview';
+import { FileDeleteButtonIcon } from './file_delete_button_icon';
+
+interface FileAttachmentEventProps {
+  file: FileJSON;
+}
+
+const FileAttachmentEvent = ({ file }: FileAttachmentEventProps) => {
+  const { isPreviewVisible, showPreview, closePreview } = useFilePreview();
+
+  return (
+    <>
+      {i18n.ADDED}
+      <FileNameLink file={file} showPreview={showPreview} />
+      {isPreviewVisible && <FilePreview closePreview={closePreview} selectedFile={file} />}
+    </>
+  );
+};
+
+FileAttachmentEvent.displayName = 'FileAttachmentEvent';
+
+function getFileDownloadButtonIcon(fileId: string) {
+  return <FileDownloadButtonIcon fileId={fileId} />;
+}
+
+function getFileDeleteButtonIcon(caseId: string, fileId: string) {
+  return <FileDeleteButtonIcon caseId={caseId} fileId={fileId} />;
+}
+
+const getFileAttachmentActions = ({ caseId, fileId }: { caseId: string; fileId: string }) => [
+  {
+    type: AttachmentActionType.CUSTOM as const,
+    isPrimary: true,
+    render: () => getFileDownloadButtonIcon(fileId),
+    label: i18n.DOWNLOAD_FILE,
+  },
+  {
+    type: AttachmentActionType.CUSTOM as const,
+    isPrimary: true,
+    render: () => getFileDeleteButtonIcon(caseId, fileId),
+    label: i18n.DELETE_FILE,
+  },
+];
+
+const getFileAttachmentViewObject = (props: ExternalReferenceAttachmentViewProps) => {
+  if (!isValidFileExternalReferenceMetadata(props.externalReferenceMetadata)) {
+    return {
+      type: 'regular',
+      event: i18n.ADDED_UNKNOWN_FILE,
```

Comment:
Will this render ok as a string or does it need to be a react node?

Reply:
This renders ok

<img width="1052" alt="Screenshot 2023-04-13 at 10 33 16" src="https://user-images.githubusercontent.com/1533137/231702382-32cd658a-8383-4365-9abb-d79465f1ea36.png">